contracts/src/AgentExecutor.sol

```diff
@@ -7,21 +7,33 @@ import {SubstrateTypes} from "./SubstrateTypes.sol";
 
 import {IERC20} from "./interfaces/IERC20.sol";
 import {SafeTokenTransfer, SafeNativeTransfer} from "./utils/SafeTransfer.sol";
+import {ERC20} from "./ERC20.sol";
+import {Gateway} from "./Gateway.sol";
 
 /// @title Code which will run within an `Agent` using `delegatecall`.
 /// @dev This is a singleton contract, meaning that all agents will execute the same code.
 contract AgentExecutor {
     using SafeTokenTransfer for IERC20;
     using SafeNativeTransfer for address payable;
 
+    // Emitted when token minted
+    event TokenMinted(bytes32 indexed tokenID, address token, address recipient, uint256 amount);
+
     /// @dev Execute a message which originated from the Polkadot side of the bridge. In other terms,
     /// the `data` parameter is constructed by the BridgeHub parachain.
     ///
-    function execute(bytes memory data) external {
+    function execute(bytes32 agentID, bytes memory data) external {
```

Comment:
See https://github.com/Snowfork/snowbridge/pull/1155/files#r1531673672 for an idea to experiment with

```suggestion
    function execute(bytes32 agentID, bytes memory data) external returns (bytes memory) {
```